### PR TITLE
fix(vue-base): making the vue component imports html friendly

### LINKED ISSUE #394: Append prefix for the component names in Vue
Currently, in Vue, we are just importing the component and not appending any prefixes while using them. This works fine most of the time unless the name of the component collides with an HTML tag.

In Angular we append app so it generates

``<app-header></app-header>``

but in Vue it generates

``<header></header>`` which is works as an HTML tag instead of the component.

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
fixes #394

### CODE REVIEW

File: packages/teleport-plugin-vue-base-component/src/index.ts [hunk 1]
```diff
@@ -52,7 +53,8 @@ export const createVueComponentPlugin: ComponentPluginFactory<VueComponentConfig
           const iterator = useIndex ? `(${iteratorName}, index)` : iteratorName
           return `${iterator} in ${iteratedCollection}`
         },
-        customElementTagName: (value) => value,
+        customElementTagName: (value) => UIDLUtils.createWebComponentFriendlyName(value),
+        customElementImportName: (value) => `App${value}`,
```
Comment (alexnm) [inline packages/teleport-plugin-vue-base-component/src/index.ts]: I wouldn't create a separate import name as the two need to be in sync to work

Comment (alexnm) [reply packages/teleport-plugin-vue-base-component/src/index.ts]: what I mean is that you can't have two different functions, the difference between the tag name and the import name is the camel case or dash case

[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/teleport-plugin-vue-base-component/src/index.ts]: Yeah, noticed these cases now. Checking those

**Comment by JayaKrishnaNamburu** [comment]
Tests are breaking because, the dependencies are mentioned as local in the tests. If we change that, they will be all right. @alexnm please check

File: packages/teleport-plugin-vue-base-component/src/index.ts [hunk 2]
```diff
@@ -52,7 +53,12 @@ export const createVueComponentPlugin: ComponentPluginFactory<VueComponentConfig
           const iterator = useIndex ? `(${iteratorName}, index)` : iteratorName
           return `${iterator} in ${iteratedCollection}`
         },
-        customElementTagName: (value) => value,
+        customElementTagName: (value) => {
+          return {
+            tagName: UIDLUtils.createWebComponentFriendlyName(value),
+            importName: UIDLUtils.createWebComponentFriendlyImportName(value),
```
Comment (alexnm) [inline packages/teleport-plugin-vue-base-component/src/index.ts]: can't importName be: dashToCamelCase(tagname)?

**Comment by alexnm** [comment]
closing this as it was fixed on master

**Pull request closed without merging**